src/preamble/02_what_is_a_module_and_a_package.md

```diff
@@ -1 +1,194 @@
 # What is a module and a package?
+
+## TL;DR
+
+A **module** is a unit of code defined by GHC and combined by Cabal to create
+packages. A **package** is either a library (a re-usable collection of modules),
+or an executable (an application that is the combination of one or more modules
+into a binary executable). A test suite is kind of like a package, in the sense
+that it is specified separately at the top-level of a `<project-name>.cabal`
+file, and is built up out of modules. However, it is different in the sense that
+it is usually not exposed and distributed. Broadly speaking, the point of Cabal
+is to define packages.
+
+## What is a module?
+
+At its most basic, a module is a namespace for collecting types, functions, and
+typeclasses (referred to as entities). Modules are a common feature in most
+languages, and the ML family of languages in particular are known for having a
+very robust module system (OCaml has full blown ML modules, and Haskell has
```

Comment:
a link to what "ML modules" means here would be appreciated, so the  below text and comment about `backpack` can be properly motivated 😄